app/root/header/Header.tsx

```diff
@@ -66,10 +68,33 @@ function NavDropDownButton({
 }
 
 export function Header() {
+  const isMobileWidth = useWindowSize().width < 1024
   const email = useEmail()
   const idp = useUserIdp()
   const [expanded, setExpanded] = useState(false)
   const [userMenuIsOpen, setUserMenuIsOpen] = useState(false)
+  const menuRef = useRef<HTMLDivElement>(null)
+  const userMenuRef = useRef<HTMLDivElement>(null)
+
+  const [mobile, setMobile] = useState(isMobileWidth ?? false)
+
+  useEffect(() => {
+    setMobile(isMobileWidth)
+  }, [isMobileWidth])
+
+  useOnClickOutside(menuRef, () => {
+    if (expanded) setExpanded(false)
+  })
+
+  useOnClickOutside(userMenuRef, (event) => {
+    if (
+      !document
+        .querySelector('#userMenuDropdown')
+        ?.contains(event.target as Node) &&
+      userMenuIsOpen
+    )
```

Comment:
What is this supposed to do?

Reply:
this is so clicking the user menu dropdown while it's open collapses the menu; the onClickOutside would set the state to false before the onClick toggled the state otherwise